### PR TITLE
UX: Add auto focus to hamburger and user menu dropdowns

### PR DESCRIPTION
**Pull request by eviltrout**

### CODE REVIEW

**Review by pmusaraj — APPROVED** [review]

File: app/assets/javascripts/discourse/app/widgets/header.js [hunk 1]
```diff
@@ -506,11 +506,27 @@ export default createWidget("header", {
 
     this.state.userVisible = !this.state.userVisible;
     this.toggleBodyScrolling(this.state.userVisible);
+    //
+    // auto focus on first button in dropdown
+    schedule("afterRender", () => {
+      let elem = document.querySelector(".user-menu button");
+      if (elem) {
+        elem.focus();
+      }
```
Comment (CvX) [inline app/assets/javascripts/discourse/app/widgets/header.js:515]: Did we sort out the `?.` operator situation? That's a great place to use it. 😃

Comment (jjaffeux) [reply app/assets/javascripts/discourse/app/widgets/header.js:515]: That shouod be very easy now I guess? just one babel plugin to add

Comment (pmusaraj) [reply app/assets/javascripts/discourse/app/widgets/header.js:515]: It is already supported, the only step where it was previously failing was in minification/compression with `uglifyjs`, but we've switched to `terser` everywhere now.

Comment (eviltrout) [reply app/assets/javascripts/discourse/app/widgets/header.js]: It seems to transpile locally, let's try in CI!

**Review by SamSaffron — APPROVED** [review]

File: app/assets/javascripts/discourse/app/widgets/header.js (hunk 1 above)
Comment (jjaffeux) [reply app/assets/javascripts/discourse/app/widgets/header.js]: Ok I see I added the plugin when I moved to babel 7, but thought I add removed it when we could use minification yet, but we just stopped using it in code, the plugin was still there, cool :+1:

**Pull request merged**